test/functional/rpc_getorphantxs.py

```diff
@@ -125,6 +127,33 @@ def orphan_details_match(self, orphan, tx, verbosity):
             self.log.info("Check the transaction hex of orphan")
             assert_equal(orphan["hex"], tx["hex"])
 
+    def test_max_orphan_amount(self):
+        self.log.info("Check that once we reach the max orphan amount we dont get any new orphans")
+        node = self.nodes[0]
+
+        peer_1 = node.add_p2p_connection(P2PInterface())
+
+        self.log.info("Filling up orphanage with " + str(MAX_ORPHANS) + "(MAX_ORPHANS) orphans")
+        orphans = []
+        for i in range(MAX_ORPHANS):
+            tx_parent_1 = self.wallet.create_self_transfer()
+            tx_child_1 = self.wallet.create_self_transfer(utxo_to_spend=tx_parent_1["new_utxo"])
+            orphans.append(tx_child_1["tx"])
+            peer_1.send_message(msg_tx(tx_child_1["tx"]))
+
+        peer_1.sync_with_ping()
+        orphanage = node.getorphantxs()
+        assert_equal(len(orphanage), MAX_ORPHANS)
+
+        for orphan in orphans:
+            assert tx_in_orphanage(node, orphan)
+
+        self.log.info("Check that we do not add more than the max orphan amount and that the first orphan has been dropped")
```

Comment:
The comment here says that the first orphan has been dropped and the PR title says "a random orphan gets dropped". I don't know so much about it, what is the expected behavior?